### PR TITLE
refactor(console): form validations on sign-up and sign-in page

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- use the React-Hook-Form's `useFieldArray` API to refactor `<SignInEditBox />` to support array field level error.
- move the refreshing sign-in methods states logic into the `<SignUpForm />` component rather than in the `<SignInEditBox />`

## Testing

form-validation-01

image

image

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignUpForm.tsx [hunk 1]
```diff
@@ -53,6 +60,53 @@ const SignUpForm = () => {
...
+            identifier,
+            isSignUpPasswordRequired,
+            password
+          ),
+          verificationCode: getSignInMethodVerificationCodeCheckState(identifier),
+        };
+      })
+    );
+
+    // Note: need to revalidate the sign-in method if we after we try to submit
```
Comment (wangsijie) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignUpForm.tsx:104]: type?

**Review by wangsijie — APPROVED** [review]

**Review by wangsijie — COMMENTED** [review]
Wondering is it possible to use watch?

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignUpForm.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignUpForm.tsx]: Updated.

**Comment by xiaoyijun** [comment]
> Wondering is it possible to use watch?
It's difficult to handle a value change logic in an array.

**Pull request merged**